.github/workflows/development-environment.yml

```diff
@@ -31,32 +32,27 @@ jobs:
     timeout-minutes: 5
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - uses: astral-sh/setup-uv@884ad927a57e558e7a70b92f2bccf9198a4be546 # v6
+        with:
+          version: '0.8.2'
+          # we just cache the venv-dir directly in action-setup-venv
+          enable-cache: false
+
       - uses: getsentry/action-setup-venv@3a832a9604b3e1a4202ae559248f26867b467cc7 # v2.1.1
         with:
           python-version: 3.13.1
-          cache-dependency-path: |
-            requirements-dev.txt
-            requirements-dev-frozen.txt
-          install-cmd: pip install -r requirements-dev.txt -c requirements-dev-frozen.txt
+          cache-dependency-path: uv.lock
+          install-cmd: uv sync --extra dev --frozen
+
+      - name: devenv sync
+        run: |
+          devenv --nocoderoot sync
+
       - name: test-tools
         run: make test-tools
       - name: Handle artifacts
         uses: ./.github/actions/artifacts
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           commit_sha: ${{ github.event.pull_request.head.sha }}
-  devenv:
-    runs-on: ubuntu-24.04
-    timeout-minutes: 5
-    steps:
-      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
-      - uses: getsentry/action-setup-venv@3a832a9604b3e1a4202ae559248f26867b467cc7 # v2.1.1
-        with:
-          python-version: 3.13.1
-          cache-dependency-path: |
-            requirements-dev.txt
-            requirements-dev-frozen.txt
-          install-cmd: pip install -r requirements-dev.txt -c requirements-dev-frozen.txt
-      - name: devenv sync
-        run: |
-          devenv --nocoderoot sync
```

Comment:
I assume this comes back later or ?

Reply:
so...

- getsentry needs `sentry[getsentry,dev]` (there's no way to get uv in getsentry to install sentry devdeps in a dep group, it has to be in extras)
- the only analogue of `pip install -r requirements-dev.txt` is `uv sync --only-dev` but that only works for dep groups, you can't only install an extra without installing the base project
- only solution i can think of is to have both and keep them mirrored with a script: https://github.com/getsentry/sentry/pull/96782/commits/d1411b31e2671f411bdab5f81a3304d02fabeda1#diff-dd488a1d6386beb0d1878357cb11853762cf0a3f4797aea2249d4c02424ba053R1
- python stdlib can't write toml files 😭 lol

if we want this then the last way i can think of is for that script to spit out requirements-dev.txt and then `uv pip compile` it

thoughts?

Reply:
oh hmm it's also just a file so i can simply read lines modify and write back

Reply:
note that this is not a problem with any other repo since over there we can just define all dev deps as the dependency group, and not in extras

Reply:
update: i'm going to work on deleting getsentry local dev .venv which will also let us just use --dev (dependency group) deps in sentry